### PR TITLE
Rest api implementation

### PR DESCRIPTION
**Pull request by rolandKoczan**
Logger implementations

### CODE REVIEW

File: db/migrations/1-db-init.js [hunk 1]
```diff
@@ -1,31 +1,37 @@
...
+    });
+    await knex.schema.createTable(contributionTable, (table) => {
+      table.integer('user').references('id').inTable('usr').notNullable();
+      table.integer('repository').references('id').inTable('repository').notNullable();
+      table.integer('line_count');
+      table.unique(['user', 'repository']);
+    });
+    logger.info('Migration was successful!');
+  } catch (error) {
+    logger.warn(error, 'Failed to migarte DB');
```
Comment (simonzsolt) [inline db/migrations/1-db-init.js]: Shouldn't error messages have `error` as log level?

File: index.js [hunk 2]
```diff
@@ -1,17 +1,21 @@
 const express = require('express');
 
+const logger = require('./logger');
+const httpLogger = require('express-pino-logger')({ logger: logger });
```
Comment (simonzsolt) [inline index.js]:
```suggestion
const httpLogger = require('express-pino-logger')({ logger });
```

File: index.js [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
 
 const app = express();
 
 app.use(express.json());
-app.use('/', router);
+app.use(httpLogger);
+app.use('/api/v1/', router);
 app.use((error, req, res, next) => {
     const statusCode = error.statusCode || 500;
+    logger.warn(error.toString());
```
Comment (simonzsolt) [inline index.js]: Also `error` log level.

File: index.js [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
     return res.status(statusCode).json({ error: error.toString() });
 });
 
-app.listen(config.port, () => console.log(`App is running on ${config.port}`));
+app.listen(config.port, () => logger.info(`App is running on ${config.port}`));
```
Comment (simonzsolt) [inline index.js]:
```suggestion
app.listen(config.port, () => logger.info(`App is listening on port: ${config.port}`));
```

**Review by simonzsolt — CHANGES_REQUESTED** [review]

**Review by kubischj — APPROVED** [review]

**Review by simonzsolt — APPROVED** [review]

**Pull request merged**